luigi/task.py

```diff
@@ -253,14 +255,20 @@ def __init__(self, *args, **kwargs):
         self.param_args = tuple(value for key, value in param_values)
         self.param_kwargs = dict(param_values)
 
-        # Build up task id
-        task_id_parts = []
-        param_objs = dict(params)
-        for param_name, param_value in param_values:
-            if param_objs[param_name].significant:
-                task_id_parts.append('%s=%s' % (param_name, param_objs[param_name].serialize(param_value)))
+        # task_id is a concatenation of task family, the first values of the first 3 parameters
+        # sorted by parameter name and a md5hash of the family/parameters as a cananocalised json.
+        TASK_ID_INCLUDE_PARAMS = 3
+        TASK_ID_TRUNCATE_PARAMS = 16
+        TASK_ID_TRUNCATE_HASH = 10
+
+        params = self.to_str_params(only_significant=True)
+        param_str = json.dumps(params, separators=(',', ':'), sort_keys=True)
+        param_hash = hashlib.md5(param_str.encode('utf-8')).hexdigest()
+
+        param_summary = '_'.join(p[:TASK_ID_TRUNCATE_PARAMS]
```

Comment:
I've removed references in `index.html` and `sge.py`.  Shout if you know of any more.